connectors/s3-tests/src/test/resources/api_test.bal

```diff
@@ -0,0 +1,286 @@
+            var response = amazonS3Client->deleteBucket(untaint bucketName);
+            if (response is error) {
+                createAndSendErrorResponse(caller, untaint <string>response.detail().message, 
+                                        BUCKET_DELETION_ERROR_MSG);
+            } else {
+                // If there is no error, then bucket deleted successfully. Send the success response.
+                backendResponse.setTextPayload(untaint string `${bucketName} deleted from Amazon S3.`, 
+                                        contentType = "text/plain");
+                respondAndHandleError(caller, backendResponse, RESPOND_ERROR_MSG);
+            }
+        } else {
+            createAndSendErrorResponse(caller, <string>amazonS3Client.detail().message, CLIENT_CREATION_ERROR_MSG);
+        }
+    }
+}
+
+// Function to extract the object content from request payload
+function extractRequestContent(http:Request request) returns string|xml|json|byte[]|error {
+    string contentTypeStr = request.getContentType();
+    if (contentTypeStr.equalsIgnoreCase("application/json")) {
+        var jsonObjectContent = request.getJsonPayload();
+        if (jsonObjectContent is json) {
+            return jsonObjectContent;
+        }
+    }
+    if (contentTypeStr.equalsIgnoreCase("application/xml")) {
+        var xmlObjectContent = request.getXmlPayload();
+        if (xmlObjectContent is xml) {
+            return xmlObjectContent;
+        }
+    }
+    if (contentTypeStr.equalsIgnoreCase("text/plain")) {
+        var textObjectContent = request.getTextPayload();
+        if (textObjectContent is string) {
+            return textObjectContent;
+        }
+    }
+    if (contentTypeStr.equalsIgnoreCase("application/octet-stream")) {
+        var binaryObjectContent = request.getBinaryPayload();
+        if (binaryObjectContent is byte[]) {
+            return binaryObjectContent;
+        }
+    }
+    error err = error(ERROR_CODE, { message : INVALID_PAYLOAD_MSG });
+    return err;
+}
+
+// Function to set the object content from reponse payload
+function setResponsePayload(http:Response response, amazons3:S3Object s3Object) {
+    string|xml|json|byte[] objectContent = s3Object.content;
+    if (objectContent is string) {
+        response.setTextPayload(objectContent, contentType = "text/plain");
+    } else if (objectContent is xml){
+        response.setXmlPayload(objectContent, contentType = "application/xml");
+    } else if (objectContent is json) {
+        response.setJsonPayload(objectContent, contentType = "application/json");
+    } else {
+        response.setBinaryPayload(objectContent, contentType = "application/octet-stream");
+    }
+}
+
+// Function to create the error response.
+function createAndSendErrorResponse(http:Caller caller, string errorMessage, string respondErrorMsg) {
+    http:Response response = new;
+    //Set 500 status code.
+    response.statusCode = 500;
```

Comment:
```suggestion
    response.statusCode = http:STATUS_INTERNAL_SERVER_ERROR;
```

Reply:
Will change this when b7a version used in the test is upgraded to 1.0.0 and after running a full TG job. Others are fixed.